### ISSUE TITLE
BUG: Losing information when handling of `NoneType`s in `pd.json_normalize`

### ISSUE THREAD

**Issue by a-theron** [pandas-dev/pandas#58395]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
def flatten_json(data: dict):
    sr = pd.Series(data)
    df = pd.DataFrame()
    df["jsonData"] = sr
    flat = pd.json_normalize(df['jsonData'])
    return df.join(flat)

expectedData = {
    0: {'Field1': 'SomeValue'},
    1: {'Field1': None},
    2: {},
}
expectedDf = flatten_json(expectedData)
display(expectedDf)

buggedData = {
    0: {},
    1: {'Field1': None},
    2: {},
}
buggedDf = flatten_json(buggedData)
display(buggedDf)
```

### Issue Description

When using `pd.json_normalize` I expected to keep the distinction between `None` and `NaN` cells (or at the very least for the behaviour to be the same regardless of the input.

There are 2 scenarios in the example I gave:
Scenario 1 - At least one value for a specific key (resultant column in the df) in the json is not `None` or `NaN`.
In this case the cells containing `None` are preserved in the final df
![image](https://github.com/pandas-dev/pandas/assets/127726911/ad1c23ae-d143-4e82-a345-3475084c4208)

Scenario 2 - All the values for a key (resultant column in the df) in the json payload are `None` or `NaN`.
In this case the cells containing `None` are not preserved in the final df, rather `None` is converted to `NaN`
![image](https://github.com/pandas-dev/pandas/assets/127726911/cecf1ce1-234e-454f-9816-35333eeb7b2f)

The handling of `None` should at the very least be consistent in both scenarios. In my opinion it should retain as much information as possible, so the output of scenario 1 makes the most sense to me.

### Expected Behavior

I expect to still see a `NoneType` value even when all the cells are either `None` or `NaN`
![image](https://github.com/pandas-dev/pandas/assets/127726911/a7104ac5-08a7-48a3-83bc-75d56625f3bd)

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.11.9.final.0
python-bits           : 64
OS                    : Windows
OS-release            : 10
Version               : 10.0.22631
machine               : AMD64
processor             : Intel64 Family 6 Model 140 Stepping 1, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : None
LOCALE                : English_Australia.1252

pandas                : 2.2.2
numpy                 : 1.26.4
pytz                  : 2024.1
dateutil              : 2.9.0
setuptools            : 68.2.2
pip                   : 23.3.1
Cython                : None
pytest                : None
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : None
html5lib              : None
pymysql               : None
psycopg2              : None
jinja2                : None
IPython               : 8.22.2
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : None
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : None
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
pyarrow               : None
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : None
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None
</details>

**Comment by nguyen-tien-tung** [comment]
take

**Comment by a-theron** [comment]
After reviewing the documentation, I think this might not be a bug.
I think it is a result of how Pandas handles none types.
In Panda's none type [documentation](https://pandas.pydata.org/docs/user_guide/missing_data.html) it states
> pandas uses different sentinel values to represent a missing (also referred to as NA) depending on the data type.

Which means that the different none types (None, NaN, NaT) are used depending on the dtype of the column.
So when a column only has missing values (as shown in the "bug"), Pandas has no reference dtypes so all missing values are represented by the default (NaN) none type.

So correct me if I am wrong, but I believe this might actually part of the Panda's feature set, rather than a bug.

**Comment by rhshadrach** [comment]
Agreed this is not a bug - the coercion from `None` to `NaN` happens in DataFrame construction. Perhaps this is something we can get away from in the distant future. Also, on pandas 3.x with the StringDtype, the inconsistency reported above is gone and all `None` values are coerced to `NaN`.

Closing.
